addOns/zest/src/main/java/org/zaproxy/zap/extension/zest/dialogs/ZestRecordScriptDialog.java

```diff
@@ -176,6 +186,35 @@ private List<String> getSites() {
         return list;
     }
 
+    private void launchBrowser(String url, String browserName) {
+        ExtensionSelenium extSelenium =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionSelenium.class);
+        try {
+            WebDriver wd = extSelenium.getProxiedBrowserByName(browserName);
+            if (wd == null) {
+                throw new NullPointerException();
+            }
+            String zapurl = wd.getCurrentUrl();
+            wd.get(url);
+            JavascriptExecutor jsExecutor = (JavascriptExecutor) wd;
+            jsExecutor.executeScript("localStorage.setItem('localzapurl', '" + zapurl + "')");
+            jsExecutor.executeScript("localStorage.setItem('localzapenable',false)");
+        } catch (NullPointerException e) {
+            String msg;
+            if (browserName == CHROME) {
+                msg = extSelenium.getWarnMessageFailedToStart(Browser.CHROME);
```

Comment:
Pass also the exception (should no longer need the if).

Reply:
Sorry, but I am not able to understand how to pass the extension and where to pass it. Can you please share a template code?

Reply:
I'm referring to the method `getWarnMessageFailedToStart(String, Throwable)`.

Reply:
`String getWarnMessageFailedToStart(String providedBrowserId, Throwable e) `  it needs browserId too how will I get that ?

Reply:
That's the same as `browserName` all lower case.